Client/Frontend/Browser/SearchViewController.swift

```diff
@@ -367,56 +364,26 @@ class SearchViewController: SiteTableViewController, KeyboardHelperDelegate, Loa
-            switch currentSection {
-            case .searchSuggestions:
-                // heightForRowAtIndexPath is called *before* the cell is created, so to get the height,
-                // force a layout pass first.
-                suggestionCell.layoutIfNeeded()
-                return suggestionCell.frame.height
-            default:
-                return super.tableView(tableView, heightForRowAt: indexPath)
-            }
-        }
-
-        return 0
+        return super.tableView(tableView, heightForRowAt: indexPath)
     }
 
     override func tableView(_ tableView: UITableView, heightForHeaderInSection section: Int) -> CGFloat {
         return 0
     }
 
     override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
-        switch SearchListSection(rawValue: indexPath.section)! {
-        case .searchSuggestions:
-            suggestionCell.imageView?.image = searchEngines.defaultEngine.image
-            suggestionCell.imageView?.isAccessibilityElement = true
-            suggestionCell.imageView?.accessibilityLabel = String(format: .SearchSearchEngineSuggestionAccessibilityLabel, searchEngines.defaultEngine.shortName)
-            return suggestionCell
-
-        case .bookmarksAndHistory:
-            let cell = super.tableView(tableView, cellForRowAt: indexPath)
-            if let site = data[indexPath.row] {
-                if let cell = cell as? TwoLineTableViewCell {
-                    let isBookmark = site.bookmarked ?? false
-                    cell.setLines(site.title, detailText: site.url)
-                    cell.setRightBadge(isBookmark ? self.bookmarkedBadge : nil)
-                    cell.imageView?.layer.borderColor = SearchViewControllerUX.IconBorderColor.cgColor
-                    cell.imageView?.layer.borderWidth = SearchViewControllerUX.IconBorderWidth
-                    cell.imageView?.contentMode = .center
-                    cell.imageView?.setImageAndBackground(forIcon: site.icon, website: site.tileURL) { [weak cell] in
-                        cell?.imageView?.image = cell?.imageView?.image?.createScaled(CGSize(width: SearchViewControllerUX.IconSize, height: SearchViewControllerUX.IconSize))
-                    }
-                }
-            }
-            return cell
-        }
+        let cell = super.tableView(tableView, cellForRowAt: indexPath)
+        return getCellForSection(cell, for: SearchListSection(rawValue: indexPath.section)!, indexPath)
     }
 
     override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
         switch SearchListSection(rawValue: section)! {
         case .searchSuggestions:
-            return searchEngines.shouldShowSearchSuggestions && !searchQuery.looksLikeAURL() && !isPrivate ? 1 : 0
+            if let count = suggestions?.count {
+                return count < 4 ? count : 4
+            } else {
+                return 0
+            }
```

Comment:
nit: minor cleanup with guard

```suggestion
            guard let count = suggestions?.count else { return 0 }
            return count < 4 ? count : 4
```